Source/FlowEditor/Private/DetailCustomizations/FlowNode_CustomNodeBaseDetails.cpp

```diff
@@ -0,0 +1,152 @@
+
+#include "DetailCustomizations/FlowNode_CustomNodeBaseDetails.h"
+#include "FlowAsset.h"
+#include "Nodes/Route/FlowNode_CustomNodeBase.h"
+
+#include "DetailCategoryBuilder.h"
+#include "DetailWidgetRow.h"
+#include "PropertyEditing.h"
+#include "Widgets/Input/SComboBox.h"
+#include "Widgets/Text/STextBlock.h"
+#include "Widgets/SWidget.h"
+
+
+void FFlowNode_CustomNodeBaseDetails::CustomizeDetails(IDetailLayoutBuilder& DetailLayout)
+{
+	// Subclasses must override this function (and call CustomizeDetailsInternal with the localized text)
+	checkNoEntry();
+}
+
+void FFlowNode_CustomNodeBaseDetails::CustomizeDetailsInternal(IDetailLayoutBuilder& DetailLayout, const FText& CustomRowNameText, const FText& EventNameText)
+{
+	DetailLayout.GetObjectsBeingCustomized(ObjectsBeingEdited);
+
+	if (ObjectsBeingEdited[0].IsValid())
+	{
+		UFlowNode_CustomNodeBase* FlowNodeBase = CastChecked<UFlowNode_CustomNodeBase>(ObjectsBeingEdited[0]);
+		CachedEventNameSelected = MakeShared<FName>(FlowNodeBase->GetEventName());
+	}
+
+	IDetailCategoryBuilder& Category = CreateDetailCategory(DetailLayout);
+
+	Category.AddCustomRow(CustomRowNameText)
+		.NameContent()
+			[
+				SNew(STextBlock)
+					.Text(EventNameText)
+			]
+		.ValueContent()
+			.HAlign(HAlign_Fill)
+			[
+				SAssignNew(EventTextListWidget, SComboBox<TSharedPtr<FName>>)
+					.OptionsSource(&EventNames)
+					.OnGenerateWidget(this, &FFlowNode_CustomNodeBaseDetails::GenerateEventWidget)
+					.OnComboBoxOpening(this, &FFlowNode_CustomNodeBaseDetails::OnComboBoxOpening)
+					.OnSelectionChanged(this, &FFlowNode_CustomNodeBaseDetails::PinSelectionChanged)
+					[
+						SNew(STextBlock)
+							.Text(this, &FFlowNode_CustomNodeBaseDetails::GetSelectedEventText)
+					]
+			];
+}
+
+void FFlowNode_CustomNodeBaseDetails::OnComboBoxOpening()
+{
+	RebuildEventNames();
+}
+
+void FFlowNode_CustomNodeBaseDetails::RebuildEventNames()
+{
+	EventNames.Empty();
+
+	check(CachedEventNameSelected.IsValid());
+	EventNames.Add(CachedEventNameSelected);
+
+	if (ObjectsBeingEdited[0].IsValid() && ObjectsBeingEdited[0].Get()->GetOuter())
+	{
+		const UFlowAsset* FlowAsset = CastChecked<UFlowAsset>(ObjectsBeingEdited[0].Get()->GetOuter());
+		TArray<FName> SortedNames = BuildEventNames(*FlowAsset);
+
+		if (bExcludeReferencedEvents)
+		{
+			for (const TPair<FGuid, UFlowNode*>& Node : FlowAsset->GetNodes())
+			{
+				if (Node.Value->GetClass()->IsChildOf(UFlowNode_CustomNodeBase::StaticClass()))
+				{
+					SortedNames.Remove(Cast<UFlowNode_CustomNodeBase>(Node.Value)->GetEventName());
+				}
+			}
+		}
+
+		SortedNames.Sort([](const FName& A, const FName& B)
+		{
+			return A.LexicalLess(B);
+		});
+
+		for (const FName& EventName : SortedNames)
+		{
+			const bool bIsCurrentSelection = (EventName == *CachedEventNameSelected);
+			if (!EventName.IsNone() && !bIsCurrentSelection)
+			{
+				EventNames.Add(MakeShared<FName>(EventName));
+			}
+		}
+	}
+
+	if (!IsInEventNames(NAME_None))
```

Comment:
_It's possible that the current selection is "None".  In that case only, the None is at the top of the combo box list.  This is consistent with the other cases where the 'current' selection is always forced to the top of the list._